zio-http/src/test/scala/zhttp/http/EncodeRequestSpec.scala

```diff
@@ -0,0 +1,85 @@
+package zhttp.http
+
+import io.netty.handler.codec.http.{HttpHeaderNames, HttpVersion}
+import zhttp.internal.HttpGen
+import zhttp.service.EncodeClientParams
+import zio.random.Random
+import zio.test.Assertion._
+import zio.test._
+
+object EncodeRequestSpec extends DefaultRunnableSpec with EncodeClientParams {
+
+  val anyClientParam: Gen[Random with Sized, Request] = HttpGen.clientRequest(
+    HttpGen.httpData(
+      Gen.listOf(Gen.alphaNumericString),
+    ),
+  )
+
+  val clientParamWithAbsoluteUrl = HttpGen.clientRequest(
+    dataGen = HttpGen.httpData(
+      Gen.listOf(Gen.alphaNumericString),
+    ),
+    urlGen = HttpGen.genAbsoluteURL,
+  )
+
+  def clientParamWithFiniteData(size: Int): Gen[Random with Sized, Request] = HttpGen.clientRequest(
```

Comment:
Should we keep it HttpGen?

Reply:
we can keep it here for now, as this test is the only user of this Gen

Reply:
This was already there, the file has been renamed only.